clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -935,6 +935,8 @@ private void initiateConnect(Node node, long now) {
         /* true iff there is a metadata request that has been sent and for which we have not yet received a response */
         private boolean metadataFetchInProgress;
 
+        private int inProgressRequestVersion;
```

Comment:
Perhaps this needs to be volatile?

Reply:
Hmm.. Can you clarify why? I think the expectations is still that NetworkClient is not thread-safe.

Reply:
I was thinking that it needs to be volatile since we can set it from the main consumer polling thread and read it from the heartbeat thread when processing metadata response.

Reply:
I could be mistaken, but I think the locking in `ConsumerNetworkClient` protects this.

Reply:
Ah yes, that makes sense.